### DISCUSSION TITLE
dynamically change collection name

### DISCUSSION THREAD

**Discussion by oprincipe** [typegoose/typegoose#442] [Q&A] [answered]
Hi all,
there's a way to change a collection name dynamically?
unfortunately doing something like this:

`
const ClassModel = getModelForClass(
        ClassRecord,
        {
            existingMongoose: mongoose,
            schemaOptions: {
                collection: collectionName
            }
        }
    )
`

alfter the first time the collection name will no longer be changed cause is stored into the internal cache.
I've also tried to use the options with:
`
options: {
                customName: my_custom_variable,
                automaticName: true
            }
`

but didn't work...

Thanks for your help

**Comment by hasezoey** [comment, marked as answer]
if at all, it should be an function (an arrow function should be enough), but does mongoose support this?

**Reply by oprincipe** [reply]
well yes, with mongoose it's quite easy tho:
`
const model = mongoose.model('Log', schema, 'log202011');
const differentModelInstance = mongoose.model('Log', schema, 'log202012');
console.log(model === differentModelInstance); // false
`

but dunno why I can't do it easily passing the schemaOption in ts...

**Reply by hasezoey** [reply]
i think related are #419 and #401

**Reply by oprincipe** [reply]
saw that... unfortunately doesn't solve the issue...
btw I sorted that out getting the schema and using the mongoose model creation instead like:

const logModel = mongoose.model("LogRecord", LogSchema, collectionName);
const log = new logModel()

thanks you anyway
appreciate that

**Reply by hasezoey** [reply]
that is why [`buildSchema`](https://typegoose.github.io/typegoose/docs/api/functions/build-schema/) and [`addModelToTypegoose`](https://typegoose.github.io/typegoose/docs/api/functions/add-model-to-typegoose) exist
